Microsoft.Toolkit.Uwp.UI.Controls/GridSplitter/GridSplitter.Data.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Represents the control that redistributes space between columns or rows of a Grid control.
+    /// </summary>
+    public partial class GridSplitter
+    {
+        // GridSplitter has special Behavior when columns are fixed
+        // If the left column is fixed, splitter will only resize that column
+        // Else if the right column is fixed, splitter will only resize the right column
+        private enum SplitBehavior
+        {
+            Split, // Both columns/rows are star lengths
+            Resize1, // resize 1 only
+            Resize2, // resize 2 only
+        }
+
+        // Only store resize data if we are resizing
+        private class ResizeData
```

Comment:
My pleasure..Please let us know when you have an updated version,
In the meantime I'm checking with our lawyers to understand what we can and cannot do with reference code from WPF